qcodes/instrument/base.py

```diff
@@ -74,6 +78,15 @@ def __init__(self, name: str, metadata: Optional[Mapping[Any, Any]] = None) -> N
 
         self.log = get_instrument_logger(self, __name__)
 
+        # NB this attribute should not be added here otherwise full control over when
+        # the parameters are instantiated is not possible in sub classes.
+        # self._call_add_params_from_decorated_methods = True
```

Comment:
i'm a bit confused - so in order to make the decorator actually work and add parameters from decorated methods, one does need to say ``self._call_add_params_from_decorated_methods = True`` before calling the super class' init method (or explicitly call ``_add_params_from_decorated_methods`` in the init), right? i don't see that happening in all the examples in the new_style.py file, and i also think that this is an important behavior that should be documented (maybe it is already, and i just missed it?)
```suggestion
```

Reply:
Default behavior: `InstrumentBase` calls `_add_params_from_decorated_methods` during `__init__` without the need for users to set `self._call_add_params_from_decorated_methods` nor explicitly call `_add_params_from_decorated_methods`.

Note the line:
```python
if getattr(self, "_call_add_params_from_decorated_methods", True):
    self._add_params_from_decorated_methods()
```

`InstrumentBase` has no `_call_add_params_from_decorated_methods` by default so that it is more flexible to control when exactly `_add_params_from_decorated_methods` is called (implicitly by parent classes and explicitly by subclasses). I know this explanation is not very specific and that is because I cannot recall the exact (advanced) edge-case it covers.

The behavior is now documented better in `@add_parameter` docstring.

Reply:
ah, of course, the default value of True - thank you pointing it out :)